src/odd.rs

```diff
@@ -175,4 +207,68 @@ mod tests {
         let two = Odd::new(BoxedUint::from(2u8));
         assert!(bool::from(two.is_none()));
     }
+
+    #[cfg(feature = "serde")]
+    mod serde_tests {
+        use crate::{Odd, U128, U64};
+        use alloc::string::ToString;
+        use bincode::ErrorKind;
+
+        #[test]
+        fn roundtrip() {
+            let uint = Odd::new(U64::from_u64(0x00123)).unwrap();
+            let ser = bincode::serialize(&uint).unwrap();
+            let deser = bincode::deserialize::<Odd<U64>>(&ser).unwrap();
+
+            assert_eq!(uint, deser);
+        }
+
+        #[test]
+        fn even_values_do_not_deserialize() {
+            let two = U128::from_u64(0x2);
+            let two_ser = bincode::serialize(&two).unwrap();
+            assert!(matches!(
+                *bincode::deserialize::<Odd<U128>>(&two_ser).unwrap_err(),
+                ErrorKind::Custom(mess) if mess == "invalid value: even, expected a non-zero odd value"
+            ))
+        }
+
+        #[test]
+        fn zero_does_not_deserialize() {
+            let zero = U64::ZERO;
+            let zero_ser = bincode::serialize(&zero).unwrap();
+
+            assert!(matches!(
+                *bincode::deserialize::<Odd<U64>>(&zero_ser).unwrap_err(),
+                ErrorKind::Custom(mess) if mess == "invalid value: even, expected a non-zero odd value"
+            ))
+        }
+
+        #[test]
+        fn cannot_deserialize_into_bigger_type() {
+            let three = Odd::new(U64::from_u64(0x3)).unwrap();
+            let three_ser = bincode::serialize(&three).unwrap();
+            let error_message = bincode::deserialize::<Odd<U128>>(&three_ser)
+                .unwrap_err()
+                .to_string();
+
+            assert_eq!(&error_message, "io error: unexpected end of file");
+        }
+
+        // @reviewers: I expected an error when deserializing an Odd<U128> into an Odd<U64>, instead I get a truncated number. Is this a big or known limitation?
```

Comment:
@tarcieri Forgot about this. :/

Is it expected that deserializing bigger into smaller truncates this way?

Reply:
That does seem like a possible bug. Can you file an issue?